test/test_piece_picker.cpp

```diff
@@ -753,6 +753,19 @@ TORRENT_TEST(sequential_download)
 			, i % blocks_per_piece));
 }
 
+TORRENT_TEST(sequential_range_download)
+{
+	// test sequential range download
+	auto p = setup_picker("7654321", "       ", "", "");
+	p->set_sequential_range(piece_index_t(1), piece_index_t(5));
+	auto picked = pick_pieces(p, "*******", 7 * blocks_per_piece, 0, nullptr
+		, piece_picker::sequential, empty_vector);
+	TEST_CHECK(int(picked.size()) == 5 * blocks_per_piece);
```

Comment:
it's not obvious to me that this is the result we want. We request 7 pieces, why only return 5?